src/main/DynamicConfig.ts

```diff
@@ -299,13 +299,45 @@ export class DynamicConfig implements IDynamicConfig {
     }
 
     public async getRemoteValue<T>(key: string, type?: ObjectType): Promise<T> {
-        return this.source(key).then((source: ISource) => {
-            if (source.key !== undefined) {
-                return this.getValueFromResolver<T>(source.key, 'remote', type)
-            } else {
-                throw new errors.ResolverUnavailable(key)
-            }
-        })
+        const source: ISource = await this.source(key) // get source for key
+
+        if (!source.key) {
+            throw new errors.ResolverUnavailable(key) // throw if key is undefined
+        }
+
+        const resolvedConfig: IRootConfigValue = await this.getConfig() // get the current config (the cached version, that is)
+
+        const value: T = await this.getValueFromResolver<T>(
+            source.key,
+            'remote',
+            type,
+        ) // get new remote value for key
+
+        if (value === null) {
```

Comment:
wondering if value will ever be null here, it looks like `getValueFromResolver` always returns a resolved or rejected promise, I'm thinking that this may never evaluate to true and we might be able to remove this check altogether but I might be missing something 🤔

Reply:
That's such a great point! My old `golang` habits die hard haha. I'll update that now!